### PR TITLE
Require vault to run as non root

### PR DESCRIPTION
**Pull request by jasonodonnell**
This removes the need to run as root user by adding extra security contexts and setting an emptyDir memory volume in the Vault users home directory (so Vault can cache tokens there if needed).

I've removed the `mlock` configurable: with this change if mlock is not disabled in Vault config, it will error due to not having root privileges.  Since this is running in Kubernetes already, we know that swap has been disabled.

### CODE REVIEW

File: templates/server-statefulset.yaml [hunk 1]
```diff
@@ -41,17 +41,21 @@ spec:
       terminationGracePeriodSeconds: 10
       serviceAccountName: {{ template "vault.fullname" . }}
       securityContext:
+        readOnlyRootFilesystem: true
+        runAsNonRoot: true
+        runAsGroup: 1000
+        runAsUser: 100
```
Comment (malnick) [inline templates/server-statefulset.yaml]: One thing I just realized here is if someone overrides our docker image with their own the UID and GID may differ, so we may want to either document that in the docs or comment inline here, or consider making this a template variable.

Comment (jasonodonnell) [reply templates/server-statefulset.yaml]: I can make this configurable, good point @malnick !

**Comment by ryno696** [comment]
Kk

**Comment by ryno696** [comment]
Yesbu can

**Comment by ryno696** [comment]
Yesbu can

**Pull request merged**